modules/Rubrics/src/Visualise.php

```diff
@@ -0,0 +1,168 @@
+
+    protected $columns;
+
+    protected $rows;
+
+    protected $cells;
+
+    protected $contexts;
+
+    /**
+     * Constructor
+     *
+     * @version  3rd May 2016
+     * @since    3rd May 2016
+     * @param    Gibbon\session
+     * @param    Gibbon\config
+     * @param    Gibbon\Contracts\Database\Connection
+     * @return   void
+     */
+    public function __construct(\Gibbon\Core $gibbon, Connection $pdo, $page, Int $gibbonPersonID, Array $columns, Array $rows, Array $cells, Array $contexts)
+    {
+        $this->session = $gibbon->session;
+
+        $this->pdo = $pdo;
+
+        $this->page = $page;
+
+        $this->gibbonPersonID = $gibbonPersonID;
+
+        $this->columns = $columns;
+
+        $this->rows = $rows;
+
+        $this->cells = $cells;
+
+        $this->contexts = $contexts;
+    }
+
+    public function renderVisualise($legend = true, $image = false)
+    {
+        //Filter out columns to ignore from visualisation
+        $this->columns = array_filter($this->columns, function ($item) {
+            return (isset($item['visualise']) && $item['visualise'] == 'Y');
+        });
+
+        if (!empty($this->columns) && !empty($this->cells)) {
+            //Cycle through rows to calculate means
+            $means = array() ;
+            foreach ($this->rows as $row) {
+                $means[$row['gibbonRubricRowID']]['title'] = $row['title'];
+                $means[$row['gibbonRubricRowID']]['cumulative'] = 0;
+                $means[$row['gibbonRubricRowID']]['denonimator'] = 0;
+
+                //Cycle through cells, and grab those for this row
+                $cellCount = 1 ;
+                foreach ($this->cells[$row['gibbonRubricRowID']] as $cell) {
+                    $visualise = false ;
+                    foreach ($this->columns as $column) {
+                        if ($column['gibbonRubricColumnID'] == $cell['gibbonRubricColumnID']) {
+                            $visualise = true ;
+                        }
+                    }
+
+                    if ($visualise) {
+                        foreach ($this->contexts as $entry) {
+                            if ($entry['gibbonRubricCellID'] == $cell['gibbonRubricCellID']) {
+                                $means[$row['gibbonRubricRowID']]['cumulative'] += $cellCount;
+                                $means[$row['gibbonRubricRowID']]['denonimator']++;
+                            }
+                        }
+                        $cellCount++;
+                    }
+                }
+            }
+
+            $columnCount = count($this->columns);
+            $data = array_map(function ($mean) use ($columnCount) {
+                return !empty($mean['denonimator'])
+                ? round((($mean['cumulative']/$mean['denonimator'])/$columnCount), 2)
+                : 0;
+            }, $means);
+
+            $this->page->scripts->add('chart');
+
+            $chart = Chart::create('visualisation', 'polarArea')
```

Comment:
I wonder if this class could extend the Chart class, rather than creating one internally 🤔

Reply:
This definitely makes sense, but I'm out of time for this one. Feel free to tinker, but otherwise I'll leave it as it is for now. I'll commit the changes above for merging. Thanks Sensei!